addons/account_edi/models/account_edi_format.py

```diff
@@ -19,51 +19,138 @@ class AccountEdiFormat(models.Model):
 
     name = fields.Char()
     code = fields.Char()
-    hide_on_journal = fields.Selection([('import_export', 'Import/Export'), ('import', 'Import Only')], default='import_export', help='used to hide this EDI format on journals')
 
     _sql_constraints = [
         ('unique_code', 'unique (code)', 'This code already exists')
     ]
 
+
+    ####################################################
+    # Low-level methods
+    ####################################################
+
+    @api.model_create_multi
+    def create(self, vals_list):
+        edi_formats = super().create(vals_list)
+
+        journals = self.env['account.journal'].search([])
+        for journal in journals:
+            for edi_format in edi_formats:
+                if edi_format._is_compatible_with_journal(journal):
+                    journal.edi_format_ids += edi_format
+
+        return edi_formats
+
     ####################################################
     # Export method to override based on EDI Format
     ####################################################
 
-    def _export_invoice_to_attachment(self, invoice):
-        """ Create the file content representing the invoice.
+    def _is_required_for_invoice(self, invoice):
+        """ Indicate if this EDI must be generated for the invoice passed as parameter.
+
+        :param invoice: An account.move having the invoice type.
+        :returns:       True if the EDI must be generated, False otherwise.
+        """
+        # TO OVERRIDE
```

Comment:
Remove it or put it also in the other functions ^^ Or just add it to the dosctrings.
(mutliple occurences)